spec/system/distribution_system_spec.rb

```diff
@@ -24,21 +24,26 @@
   end
 
   context "When creating a new distribution manually" do
-    it "Allows a distribution to be created" do
-      visit @url_prefix + "/distributions/new"
+    context "when the delivery_method is not shipped" do
+      it "Allows a distribution to be created and shipping cost field not visible" do
+        visit @url_prefix + "/distributions/new"
 
-      select @partner.name, from: "Partner"
-      select @storage_location.name, from: "From storage location"
-      choose "Pick up"
+        select @partner.name, from: "Partner"
+        select @storage_location.name, from: "From storage location"
+        choose "Pick up"
 
-      fill_in "Comment", with: "Take my wipes... please"
-      fill_in "Distribution date", with: '01/01/2001 10:15:00 AM'
+        fill_in "Comment", with: "Take my wipes... please"
+        fill_in "Distribution date", with: '01/01/2001 10:15:00 AM'
 
-      expect(PartnerMailerJob).to receive(:perform_later).once
-      click_button "Save", match: :first
+        # shipping cost field should not be visible
+        expect { page.find("#dist_shipping_cost") }.to raise_error(Capybara::ElementNotFound)
```

Comment:
You'll want `find_by_id` here to return `nil` rather than waiting for an error. Also make sure to pass a shorter wait time so it doesn't hang for 10 seconds or however long.

Reply:
I checked find_by_id also raise exception if element not found, but still used find_by_id with wait:2